spark/src/main/scala/geotrellis/spark/summary/polygonal/RDDPolygonalSummary.scala

```diff
@@ -0,0 +1,143 @@
+    * @param geometryRdd
+    * @param visitor
+    * @param options Rasterizer.Options for this polygonal summary operation
+    * @param semigroupR
+    * @param tagR
+    * @tparam R Type of the summary being performed
+    * @tparam T Type of the Raster contained in rasterRdd
+    * @return An RDD containing each unique input geometry and it's summary result R
+    */
+  def apply[R, T <: CellGrid[Int]](
+      rasterRdd: RDD[(SpatialKey, T)] with Metadata[TileLayerMetadata[SpatialKey]],
+      geometryRdd: RDD[Geometry],
+      visitor: GridVisitor[Raster[T], R],
+      options: Rasterizer.Options
+  )(implicit semigroupR: Semigroup[R],
+    tagR: ClassTag[R]): RDD[Feature[Geometry, PolygonalSummaryResult[R]]] = {
+    val layout = rasterRdd.metadata.layout
+
+    val keyedGeometryRdd: RDD[(SpatialKey, Feature[Geometry, UUID])] =
+      geometryRdd.flatMap { geom =>
+        val uuid = UUID.randomUUID
+        val keys: Set[SpatialKey] = layout.mapTransform.keysForGeometry(geom)
+        keys.map(k => (k, Feature(geom, uuid)))
+      }
+
+    // Combine geometries and rasters by SpatialKey, then loop partitions
+    //  and perform the polygonal summary on each combination
+    val joinedRdd = keyedGeometryRdd.join(rasterRdd)
+    val featuresWithSummaries: RDD[(UUID, Feature[Geometry, PolygonalSummaryResult[R]])] =
+      joinedRdd.map { data => {
+        val spatialKey = data._1
+        val feature: Feature[Geometry, UUID] = data._2._1
+        val tile: T = data._2._2
+        val extent: Extent = layout.mapTransform.keyToExtent(spatialKey)
+        val raster: Raster[T] = Raster(tile, extent)
+        val result: PolygonalSummaryResult[R] =
+          raster.polygonalSummary(feature.geom, visitor.getClass.newInstance, options)
+        (feature.data, Feature(feature.geom, result))
+      }
+    }
+
+    // Geometries may have been split across SpatialKey leading to multiple UUID keys.
+    // Combine by UUID and then drop the UUID, which was added within this method
+    featuresWithSummaries
+      .reduceByKey { (a, b) => Feature(a.geom, a.data.combine(b.data)) }
+      .map { _._2 }
+  }
+
+  trait RDDPolygonalSummaryMethods[R, T <: CellGrid[Int]]
+      extends MethodExtensions[RDD[(SpatialKey, T)] with Metadata[TileLayerMetadata[SpatialKey]]] {
+    def polygonalSummary[R](geometryRdd: RDD[Geometry],
+                            visitor: GridVisitor[Raster[T], R],
+                            options: Rasterizer.Options)(
+        implicit semigroupR: Semigroup[R], tagR: ClassTag[R]): RDD[Feature[Geometry, PolygonalSummaryResult[R]]] = {
+      RDDPolygonalSummary(self, geometryRdd, visitor, options)
+    }
+
+    def polygonalSummary[R](geometries: Seq[Geometry],
+                            visitor: GridVisitor[Raster[T], R],
+                            options: Rasterizer.Options)(
+        implicit semigroupR: Semigroup[R], tagR: ClassTag[R], sc: SparkContext): RDD[Feature[Geometry, PolygonalSummaryResult[R]]] = {
```

Comment:
This signature can be simplified by removing `sc: SparkContext`
Each `RDD` instance has `sparkContext` reference. We've been using this trick more consistently and the result has been tighter function signatures, which is good all around.

Reply:
TIL.